### PR TITLE
Fixed missing required info in resource summary

### LINKED ISSUE #1161: CLI frictionless `summary` ignores `required` property of fields
# Overview

`frictionless summary countries.resource.yaml` command ignores real value field property `required`

# Input data

File `COUNTRIES.csv`:
```csv
CID;ECC;CCD;CNAME
11;E2;2;Czech Republic
```

File `countries.resource.yaml`:
```yaml
# --------
# metadata: COUNTRIES.csv
# --------

path: COUNTRIES.csv
name: countries
profile: tabular-data-resource
scheme: file
format: csv
hashing: md5
encoding: utf-8-sig
dialect:
  delimiter: ;
schema:
  fields:
    - type: integer
      name: CID
      required: true
    - type: string
      name: ECC
      required: false
    - type: integer
      name: CCD
    - type: string
      name: CNAME
```
# Summary output
```shell
$ frictionless summary countries.resource.yaml

# Describe 

+--------+---------+------------+
| name   | type    | required   |
+========+=========+============+
| CID    | integer |            |
+--------+---------+------------+
| ECC    | string  |            |
+--------+---------+------------+
| CCD    | integer |            |
+--------+---------+------------+
| CNAME  | string  |            |
+--------+---------+------------+

# Extract 

+-----+------+-----+------------------+
| CID | ECC  | CCD | CNAME            |
+=====+======+=====+==================+
|  11 | 'E2' |   2 | 'Czech Republic' |
+-----+------+-----+------------------+


# Validate 

# -----
# valid: COUNTRIES.csv 
# -----

## Summary 

+------------------------+-------------------+
| Description            | Size/Name/Count   |
+========================+===================+
| File name              | COUNTRIES.csv     |
+------------------------+-------------------+
| File size (bytes)      | 46                |
+------------------------+-------------------+
| Total Time Taken (sec) | 0.008             |
+------------------------+-------------------+
```
As can be seen, `required` columns is empty, regardless of the fact:
- field `CID` is explicitly required
- field `ECC` is explicitly not required
- fields `CCD` and `CNAME` shall be by definition (as default value) required

### PR DESCRIPTION
**Pull request by shashigharti**
Fixed missing info of required property in resource summary
Added tests

- fixes #1161

---

Please make sure that all the checks pass. Please add here any additional information regarding this pull request. It's highly recommended that you link this PR to an issue (please create one if it doesn't exist for this PR)

### CODE REVIEW

**Comment by shashigharti** [comment]
@roll this is ready to review!

**Review by roll — CHANGES_REQUESTED** [review]
Hi @shashigharti,

I think the initial issue is invalid as there is no `field.required` property in the standards. It should be provided as `field.constraint.required`.

Can you please:
- keep all the work you done in this PR it's anyway useful changes
- but switch the required setting in tests to a proper one (`field.constraints.required`)
- WDYT if we emit a validation error if `field.required` is provided as we did for `resource.fields` etc?

File: frictionless/schema/schema.py [hunk 1]
```diff
@@ -262,6 +262,10 @@ def to_excel_template(self, path: str):
 
     def to_summary(self) -> str:
         """Summary of the schema in table format"""
+        # Setting required property of the field which is included as custom fields
```
Comment (roll) [inline frictionless/schema/schema.py]: So we need to remove these 4 lines

Comment (shashigharti) [reply frictionless/schema/schema.py]: Now I understand, I will remove these lines. I should have checked the standards. Sure adding "validation error" would be better. I will add that.

**Comment by shashigharti** [comment]
Thank you @roll . I have made the changes and it is ready to review!

File: frictionless/schema/field.py [hunk 2]
```diff
@@ -218,6 +218,13 @@ def metadata_import(cls, descriptor):
         return super().metadata_import(descriptor)
 
     def metadata_validate(self):
+        invalid_fields = {"required": "field->constraints"}
```
Comment (roll) [inline frictionless/schema/field.py]: @shashigharti 
Looks good but I slightly simplified these checks during the v5 work. Can you please follow this pattern?

> e.g. resource.py

```
        # Custom
        for name in ["missingValues", "fields"]:
            if name in self.custom:
                note = f'"{name}" should be set as "schema.{name}"'
                yield errors.ResourceError(note=note)
```

**Review by roll — CHANGES_REQUESTED** [review]

File: frictionless/schema/field.py (hunk 2 above)
Comment (shashigharti) [reply frictionless/schema/field.py]: Thank you!. I had checked that but got confused. I will fix that.

**Comment by shashigharti** [comment]
@roll thank you! I have made the changes. It is ready for the review!

**Review by roll — APPROVED** [review]
Thanks!

**Pull request merged**